src/test/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalServiceTest.kt

```diff
@@ -3,60 +3,71 @@ package uk.gov.justice.digital.hmpps.sentenceplan.services
-  private val goalService = GoalService(goalRepository, stepRepository, stepActorRepository, areaOfNeedRepository)
-  private val uuid = UUID.fromString("ef74ee4b-5a0b-481b-860f-19187260f2e7")
-
-  private val stepEntity = StepEntity(
-    description = "description",
-    id = 123L,
-    status = "status",
-    relatedGoalUuid = UUID.fromString("ef74ee4b-5a0b-481b-860f-19187260f2e7"),
-    creationDate = currentTime,
-  )
-  private val actorsEntityList = listOf(
-    StepActorsEntity(1, UUID.fromString("71793b64-545e-4ae7-9936-610639093857"), "actor", 1),
+  private val planRepository: PlanRepository = mockk()
+  private val goalService = GoalService(goalRepository, areaOfNeedRepository, planRepository)
+  private val goalUuid = UUID.fromString("ef74ee4b-5a0b-481b-860f-19187260f2e7")
+
+  private val goalEntityNoSteps: GoalEntity = GoalEntity(
+    title = "Mock Goal",
+    areaOfNeedUuid = UUID.randomUUID(),
+    plan = null,
+    uuid = goalUuid,
   )
+
   private val actors = listOf(
-    StepActor("actor", 1),
+    StepActor("actor 1", 1),
+    StepActor("actor 2", 2),
   )
-  private val steps = Step(
-    description = "description",
-    status = "status",
-    actor = actors,
+  private val steps = listOf(
+    Step(
+      description = "description 1",
+      status = "status 1",
+      actor = actors,
+    ),
+    Step(
+      description = "description 2",
+      status = "status 2",
+      actor = actors,
+    ),
   )
 
   @Test
   fun `create new steps`() {
-    every { stepRepository.save(any<StepEntity>()) } returns stepEntity
-    every { stepActorRepository.saveAll(any<List<StepActorsEntity>>()) } returns actorsEntityList
-    val stepsList = goalService.createNewSteps(uuid, listOf(steps))
-    assertThat(stepsList.get(0).status).isEqualTo("status")
-    assertThat(stepsList.get(0).id).isEqualTo(123)
-    assertThat(stepsList.get(0).relatedGoalUuid).isEqualTo(uuid)
-    assertThat(stepsList.get(0).description).isEqualTo("description")
-    assertThat(stepsList.get(0).creationDate).isEqualTo(currentTime)
+    val goalSlot = slot<GoalEntity>()
+    every { goalRepository.findByUuid(goalUuid) } returns goalEntityNoSteps
+    every { goalRepository.save(capture(goalSlot)) } answers { goalSlot.captured }
+
+    val stepsList = goalService.createNewSteps(goalUuid, steps).steps
+
+    assertThat(stepsList.size).isEqualTo(2)
+
+    assertThat(stepsList.get(0).status).isEqualTo("status 1")
+    assertThat(stepsList.get(0).goal?.uuid).isEqualTo(goalUuid)
+    assertThat(stepsList.get(0).description).isEqualTo("description 1")
+    assertThat(stepsList.get(0).actors.size).isEqualTo(2)
+    assertThat(stepsList.get(0).actors.get(0).actor).isEqualTo("actor 1")
+    assertThat(stepsList.get(0).actors.get(1).actor).isEqualTo("actor 2")
+
+    assertThat(stepsList.get(1).status).isEqualTo("status 2")
+    assertThat(stepsList.get(1).goal?.uuid).isEqualTo(goalUuid)
+    assertThat(stepsList.get(1).description).isEqualTo("description 2")
```

Comment:
Intellij tells me that we should use indexing e.g. `stepsList[0]` instead of `get(0)`.

Reply:
Fixed in https://github.com/ministryofjustice/hmpps-sentence-plan/pull/55/commits/040d4f52592f085634c6b6282c1411274d3cf5d3